ai-catalog/ui-component/src/main/java/com/android/ai/uicomponent/TextInput.kt

```diff
@@ -0,0 +1,95 @@
+package com.android.ai.uicomponent
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.android.ai.theme.AISampleCatalogTheme
+
+@Composable
+fun TextInput(
+    value: String,
+    modifier: Modifier = Modifier,
+    hint: String = "",
+    placeholder: String = "",
+    primaryButton: @Composable () -> Unit = {},
+    secondaryButton: @Composable () -> Unit = {},
+    onValueChange: (String) -> Unit
+) {
+    Row (
+       modifier = modifier
+           .border(
+               1.dp,
+               MaterialTheme.colorScheme.outline,
+               shape = RoundedCornerShape(30.dp)
+           )
+           .height(56.dp)
+           .clip(
+               shape = RoundedCornerShape(30.dp),
+           )
+           .background(color = MaterialTheme.colorScheme.surfaceContainerHigh)
+    ){
+        OutlinedTextField(
+            value = value,
+            label = { Text(text = hint) },
+            placeholder = { Text(text = placeholder) },
+            onValueChange = onValueChange,
+            modifier = Modifier
+                .weight(1f)
+                .height(48.dp)
+                .padding(start = 12.dp),
+            colors = TextFieldDefaults.colors(
+                focusedContainerColor = Color.Transparent,
+                unfocusedContainerColor = Color.Transparent,
+                disabledContainerColor = Color.Transparent,
+                focusedIndicatorColor = Color.Transparent,
+                unfocusedIndicatorColor = Color.Transparent,
+                disabledIndicatorColor = Color.Transparent,
```

Comment:
Whats the intention here? Maybe add a comment explaining why its all transparent.

Reply:
If you don't need the Outline or style that Material provides and need to make it custom, you should switch to BasicTextField instead.